[BUG] Terramate panics when using ternary expression with `tm_hcl_expression()` function

**Describe the bug**

Terramate seems to panic when running `generate` with a ternary expression mixed with the `tm_hcl_expression()` function.

Example:
```
globals {
  database = global.module_rds_enabled ? tm_hcl_expression("module.rds.db_instance_database_name") : tm_hcl_expression("module.rds_aurora.cluster_database_name")
}
```

**To Reproduce**

Use the next configuration
```
terramate {
  required_version = "~> 0.1.42"

  config {
    git {
      check_untracked   = false
      check_uncommitted = false
      check_remote      = false
    }
  }
}

globals {
  module_rds_enabled = true

  database = global.module_rds_enabled ? tm_hcl_expression("module.rds.db_instance_database_name") : tm_hcl_expression("module.rds_aurora.cluster_database_name")
}

generate_hcl "_test.tf" {
  content {
    database = global.database
  }
}

stack {
  name = "test-stack"
}
```

and run `terramate generate`

**Expected behavior**

The above expression should return `database = module.rds.db_instance_database_name` when `global.module_rds_enabled = true` or `database = module.rds_aurora.cluster_database_name` when `global.module_rds_enabled = false`

**Log Output**

Logs from panic:
```
panic: cannot produce tokens for customdecode.ExpressionVal(&hclsyntax.ScopeTraversalExpr{Traversal:hcl.Traversal{hcl.TraverseRoot{isTraverser:hcl.isTraverser{}, Name:"module", SrcRange:hcl.Range{Filename:"<generated-hcl>\x00module.rds_aurora.cluster_database_name", Start:hcl.Pos{Line:1, Column:1, Byte:0}, End:hcl.Pos{Line:1, Column:7, Byte:6}}}, hcl.TraverseAttr{isTraverser:hcl.isTraverser{}, Name:"rds_aurora", SrcRange:hcl.Range{Filename:"<generated-hcl>\x00module.rds_aurora.cluster_database_name", Start:hcl.Pos{Line:1, Column:7, Byte:6}, End:hcl.Pos{Line:1, Column:18, Byte:17}}}, hcl.TraverseAttr{isTraverser:hcl.isTraverser{}, Name:"cluster_database_name", SrcRange:hcl.Range{Filename:"<generated-hcl>\x00module.rds_aurora.cluster_database_name", Start:hcl.Pos{Line:1, Column:18, Byte:17}, End:hcl.Pos{Line:1, Column:40, Byte:39}}}}, SrcRange:hcl.Range{Filename:"<generated-hcl>\x00module.rds_aurora.cluster_database_name", Start:hcl.Pos{Line:1, Column:1, Byte:0}, End:hcl.Pos{Line:1, Column:40, Byte:39}}})

goroutine 1 [running]:
github.com/hashicorp/hcl/v2/hclwrite.appendTokensForValue({{{0x164a628?, 0xc00019d140?}}, {0x14aba00?, 0xc00085f170?}}, {0xc000af6c00, 0x14, 0x20})
        github.com/hashicorp/hcl/v2@v2.14.1/hclwrite/generate.go:298 +0x120b
github.com/hashicorp/hcl/v2/hclwrite.appendTokensForValue({{{0x164a820?, 0xc00085f250?}}, {0x14dac80?, 0xc000609b30?}}, {0xc000c4b680, 0x10, 0x10})
        github.com/hashicorp/hcl/v2@v2.14.1/hclwrite/generate.go:284 +0xf93
github.com/hashicorp/hcl/v2/hclwrite.appendTokensForValue({{{0x164a820?, 0xc00085f260?}}, {0x14dac80?, 0xc000609b90?}}, {0xc000c4b680, 0xc, 0x10})
        github.com/hashicorp/hcl/v2@v2.14.1/hclwrite/generate.go:284 +0xf93
github.com/hashicorp/hcl/v2/hclwrite.appendTokensForValue({{{0x164a820?, 0xc00085f280?}}, {0x14dac80?, 0xc000609bf0?}}, {0xc000025ae0, 0x4, 0x4})
        github.com/hashicorp/hcl/v2@v2.14.1/hclwrite/generate.go:284 +0xf93
github.com/hashicorp/hcl/v2/hclwrite.appendTokensForValue({{{0x164a820?, 0xc000ac4970?}}, {0x14dac80?, 0xc000bb1aa0?}}, {0x0, 0x0, 0x0})
        github.com/hashicorp/hcl/v2@v2.14.1/hclwrite/generate.go:284 +0xf93
github.com/hashicorp/hcl/v2/hclwrite.TokensForValue({{{0x164a820?, 0xc000ac4970?}}, {0x14dac80?, 0xc000bb1aa0?}})
        github.com/hashicorp/hcl/v2@v2.14.1/hclwrite/generate.go:24 +0x35
github.com/mineiros-io/terramate/hcl/eval.(*engine).evalVar(0xc000a00768)
        github.com/mineiros-io/terramate/hcl/eval/partial.go:901 +0x41a
github.com/mineiros-io/terramate/hcl/eval.(*engine).evalIdent(0xc000a00768)
        github.com/mineiros-io/terramate/hcl/eval/partial.go:609 +0x136
github.com/mineiros-io/terramate/hcl/eval.(*engine).evalExpr(0xc000a00768)
        github.com/mineiros-io/terramate/hcl/eval/partial.go:354 +0x1278
github.com/mineiros-io/terramate/hcl/eval.(*engine).Eval(0xc000a00768)
        github.com/mineiros-io/terramate/hcl/eval/partial.go:115 +0xee
github.com/mineiros-io/terramate/hcl/eval.(*Context).PartialEval(0xc0000153c8, {0x164a548?, 0xc0000ad0e0?})
        github.com/mineiros-io/terramate/hcl/eval/eval.go:126 +0x230
github.com/mineiros-io/terramate/generate/genhcl.copyBody(0xc000502090?, 0xc000426dc0, {0x164a5f0, 0xc0000153d0})
        github.com/mineiros-io/terramate/generate/genhcl/genhcl.go:310 +0x84b
github.com/mineiros-io/terramate/generate/genhcl.appendBlock(0xc000a012d0?, 0xc0001506c0, {0x164a5f0, 0xc0000153d0})
        github.com/mineiros-io/terramate/generate/genhcl/genhcl.go:338 +0xa8
github.com/mineiros-io/terramate/generate/genhcl.copyBody(0x164a740?, 0xc000426f20, {0x164a5f0, 0xc0000153d0})
        github.com/mineiros-io/terramate/generate/genhcl/genhcl.go:322 +0xa9a
github.com/mineiros-io/terramate/generate/genhcl.Load(0xc000304780?, {{0xc000037770, 0x46}, {0xc0006ed500, 0x6d, 0x6d}}, {0x164c798?, 0xc000741dc0}, 0x1?)
        github.com/mineiros-io/terramate/generate/genhcl/genhcl.go:226 +0xad4
github.com/mineiros-io/terramate/generate.loadStackCodeCfgs(0xc00055e948?, {{0xc000037770, 0x46}, {0xc0006ed500, 0x6d, 0x6d}}, 0x164c798?, 0xc000741dc0?)
        github.com/mineiros-io/terramate/generate/generate.go:963 +0xe7
github.com/mineiros-io/terramate/generate.Do.func1({{0xc000037770, 0x46}, {0xc0006ed500, 0x6d, 0x6d}}, 0xc000741dc0, 0xc0007f5e00?)
        github.com/mineiros-io/terramate/generate/generate.go:159 +0x46e
github.com/mineiros-io/terramate/generate.forEachStack(0xc000304790?, {0xc000320840, 0x56}, 0xc000a036a0)
        github.com/mineiros-io/terramate/generate/generate.go:686 +0xacd
github.com/mineiros-io/terramate/generate.Do(0xc0006fe098, {0xc000320840?, 0x1f4?})
        github.com/mineiros-io/terramate/generate/generate.go:138 +0x85
github.com/mineiros-io/terramate/cmd/terramate/cli.(*cli).generate(0xc0006fe000, {0xc000320840?, 0x0?})
        github.com/mineiros-io/terramate/cmd/terramate/cli/cli.go:600 +0x4a
github.com/mineiros-io/terramate/cmd/terramate/cli.(*cli).run(0xc0006fe000)
        github.com/mineiros-io/terramate/cmd/terramate/cli/cli.go:392 +0x4d8
github.com/mineiros-io/terramate/cmd/terramate/cli.Exec({0xc0001ac010, 0x1, 0x1}, {0x1647ca0, 0xc0001a0000}, {0x1647cc0, 0xc0001a0008}, {0x1647cc0, 0xc0001a0010})
        github.com/mineiros-io/terramate/cmd/terramate/cli/cli.go:202 +0xdd
main.main()
        github.com/mineiros-io/terramate/cmd/terramate/main.go:30 +0x78
```

**Environment (please complete the following information):**
 - OS: Mac OSX
 - OS Version: 12.6 (21G115)
 - Git Version: git version 2.37.1 (Apple Git-137.1)
 - Terramate Version: 0.1.42

--- a/docs/functions.md
+++ b/docs/functions.md
@@ -55,7 +55,14 @@ contents as an expression. It is particularly useful to circumvent some
 limitations on HCL and Terraform when building complex expressions from
 dynamic data.
 
-For example, given a global named data defined like this:
+Since this function produces an expression, not a final evaluated value,
+it is only allowed to be used on contexts where partial evaluation is
+allowed, which currently are:
+
+* `generate_hcl.content` block
+* `generate_file.content` attribute
+
+To use `tm_hcl_expression`, lets say we have a global named data defined like this:
 
 ```
 globals {
@@ -67,13 +74,17 @@ You can use this global to build a complex expression when generation code,
 like this:
 
 ```hcl
-tm_hcl_expression("data.google_active_folder._parent_id.id.${global.data}")
+generate_hcl "test.hcl" {
+    content {
+        expr = tm_hcl_expression("data.google_active_folder._parent_id.id.${global.data}")
+    }
+}
 ```
 
-Which will produce the expression:
+Which will generate:
 
 ```hcl
-data.google_active_folder._parent_id.id.data
+expr = data.google_active_folder._parent_id.id.data
 ```
 
 ## Experimental Functions

--- a/docs/functions.md
+++ b/docs/functions.md
@@ -57,10 +57,7 @@ dynamic data.
 
 Since this function produces an expression, not a final evaluated value,
 it is only allowed to be used on contexts where partial evaluation is
-allowed, which currently are:
-
-* `generate_hcl.content` block
-* `generate_file.content` attribute
+allowed, which currently is only the `generate_hcl.content` block.
 
 To use `tm_hcl_expression`, lets say we have a global named data defined like this:
 

--- a/generate/genfile/genfile.go
+++ b/generate/genfile/genfile.go
@@ -152,7 +152,7 @@ func Load(
 		vendorTargetDir := project.NewPath(path.Join(
 			sm.Path().String(),
 			path.Dir(name)))
-		evalctx.SetTmVendor(vendorTargetDir, vendorDir, vendorRequests)
+		evalctx.AddTmVendor(vendorTargetDir, vendorDir, vendorRequests)
 
 		file, err := Eval(genFileBlock, evalctx.Context)
 		if err != nil {

--- a/generate/genhcl/genhcl.go
+++ b/generate/genhcl/genhcl.go
@@ -170,7 +170,7 @@ func Load(
 		vendorTargetDir := project.NewPath(path.Join(
 			sm.Path().String(),
 			path.Dir(name)))
-		evalctx.SetTmVendor(vendorTargetDir, vendorDir, vendorRequests)
+		evalctx.AddTmVendor(vendorTargetDir, vendorDir, vendorRequests)
 
 		err := lets.Load(hclBlock.Lets, evalctx.Context)
 		if err != nil {
@@ -232,6 +232,8 @@ func Load(
 			})
 			continue
 		}
+
+		evalctx.AddTmHCLExpression()
 
 		gen := hclwrite.NewEmptyFile()
 		if err := copyBody(gen.Body(), hclBlock.Content.Body, evalctx); err != nil {

--- a/generate/hcl_expr_func_test.go
+++ b/generate/hcl_expr_func_test.go
@@ -29,7 +29,7 @@ func TestHCLExpressionFunc(t *testing.T) {
 	// In the future tests could be moved here.
 	testCodeGeneration(t, []testcase{
 		{
-			name: "tm_hcl_expression is not available",
+			name: "not available on lets block",
 			layout: []string{
 				"s:stack",
 			},

--- a/generate/hcl_expr_func_test.go
+++ b/generate/hcl_expr_func_test.go
@@ -1,0 +1,74 @@
+// Copyright 2022 Mineiros GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package generate_test
+
+import (
+	"testing"
+
+	"github.com/mineiros-io/terramate/errors"
+	"github.com/mineiros-io/terramate/generate"
+	"github.com/mineiros-io/terramate/hcl/eval"
+
+	. "github.com/mineiros-io/terramate/test/hclwrite/hclutils"
+)
+
+func TestHCLExpressionFunc(t *testing.T) {
+	// TODO(KATCIPIS): currently most behavior is tested on the genhcl pkg.
+	// In the future tests could be moved here.
+	testCodeGeneration(t, []testcase{
+		{
+			name: "tm_hcl_expression is not available",
+			layout: []string{
+				"s:stack",
+			},
+			configs: []hclconfig{
+				{
+					path: "/stack",
+					add: Doc(
+						GenerateHCL(
+							Labels("test.hcl"),
+							Lets(
+								Expr("expr", `tm_hcl_expression("test")`),
+							),
+							Content(
+								Expr("value", `let.expr`),
+							),
+						),
+						GenerateFile(
+							Labels("test.txt"),
+							Lets(
+								Expr("content", `tm_hcl_expression("test")`),
+							),
+							Expr("content", "let.content"),
+						),
+					),
+				},
+			},
+			wantReport: generate.Report{
+				Failures: []generate.FailureResult{
+					{
+						Result: generate.Result{
+							Dir: "/stack",
+						},
+						Error: errors.L(
+							errors.E(eval.ErrEval),
+							errors.E(eval.ErrEval),
+						),
+					},
+				},
+			},
+		},
+	})
+}

--- a/generate/hcl_expr_func_test.go
+++ b/generate/hcl_expr_func_test.go
@@ -29,7 +29,7 @@ func TestHCLExpressionFunc(t *testing.T) {
 	// In the future tests could be moved here.
 	testCodeGeneration(t, []testcase{
 		{
-			name: "not available on lets block",
+			name: "not available on generate_hcl lets block",
 			layout: []string{
 				"s:stack",
 			},
@@ -46,6 +46,29 @@ func TestHCLExpressionFunc(t *testing.T) {
 								Expr("value", `let.expr`),
 							),
 						),
+					),
+				},
+			},
+			wantReport: generate.Report{
+				Failures: []generate.FailureResult{
+					{
+						Result: generate.Result{
+							Dir: "/stack",
+						},
+						Error: errors.E(eval.ErrEval),
+					},
+				},
+			},
+		},
+		{
+			name: "not available on generate_file lets block",
+			layout: []string{
+				"s:stack",
+			},
+			configs: []hclconfig{
+				{
+					path: "/stack",
+					add: Doc(
 						GenerateFile(
 							Labels("test.txt"),
 							Lets(
@@ -62,10 +85,7 @@ func TestHCLExpressionFunc(t *testing.T) {
 						Result: generate.Result{
 							Dir: "/stack",
 						},
-						Error: errors.L(
-							errors.E(eval.ErrEval),
-							errors.E(eval.ErrEval),
-						),
+						Error: errors.E(eval.ErrEval),
 					},
 				},
 			},

--- a/generate/hcl_expr_func_test.go
+++ b/generate/hcl_expr_func_test.go
@@ -61,6 +61,124 @@ func TestHCLExpressionFunc(t *testing.T) {
 			},
 		},
 		{
+			name: "not available on generate_hcl condition",
+			layout: []string{
+				"s:stack",
+			},
+			configs: []hclconfig{
+				{
+					path: "/stack",
+					add: Doc(
+						GenerateHCL(
+							Labels("test.hcl"),
+							Expr("condition", `tm_hcl_expression("test")`),
+							Content(),
+						),
+					),
+				},
+			},
+			wantReport: generate.Report{
+				Failures: []generate.FailureResult{
+					{
+						Result: generate.Result{
+							Dir: "/stack",
+						},
+						Error: errors.E(eval.ErrEval),
+					},
+				},
+			},
+		},
+		{
+			name: "not available on generate_hcl assert",
+			layout: []string{
+				"s:stack",
+			},
+			configs: []hclconfig{
+				{
+					path: "/stack",
+					add: Doc(
+						GenerateHCL(
+							Labels("test.hcl"),
+							Assert(
+								Expr("assertion", `tm_hcl_expression("true")`),
+								Str("message", "msg"),
+							),
+							Content(),
+						),
+					),
+				},
+			},
+			wantReport: generate.Report{
+				Failures: []generate.FailureResult{
+					{
+						Result: generate.Result{
+							Dir: "/stack",
+						},
+						Error: errors.E(eval.ErrEval),
+					},
+				},
+			},
+		},
+		{
+			name: "not available on generate_file condition",
+			layout: []string{
+				"s:stack",
+			},
+			configs: []hclconfig{
+				{
+					path: "/stack",
+					add: Doc(
+						GenerateFile(
+							Labels("test.txt"),
+							Expr("condition", `tm_hcl_expression("test")`),
+							Str("content", "content"),
+						),
+					),
+				},
+			},
+			wantReport: generate.Report{
+				Failures: []generate.FailureResult{
+					{
+						Result: generate.Result{
+							Dir: "/stack",
+						},
+						Error: errors.E(eval.ErrEval),
+					},
+				},
+			},
+		},
+		{
+			name: "not available on generate_file assert",
+			layout: []string{
+				"s:stack",
+			},
+			configs: []hclconfig{
+				{
+					path: "/stack",
+					add: Doc(
+						GenerateFile(
+							Labels("test.txt"),
+							Assert(
+								Expr("assertion", `tm_hcl_expression("true")`),
+								Str("message", "msg"),
+							),
+							Str("content", "content"),
+						),
+					),
+				},
+			},
+			wantReport: generate.Report{
+				Failures: []generate.FailureResult{
+					{
+						Result: generate.Result{
+							Dir: "/stack",
+						},
+						Error: errors.E(eval.ErrEval),
+					},
+				},
+			},
+		},
+		{
 			name: "not available on generate_file lets block",
 			layout: []string{
 				"s:stack",
@@ -76,6 +194,32 @@ func TestHCLExpressionFunc(t *testing.T) {
 							),
 							Expr("content", "let.content"),
 						),
+					),
+				},
+			},
+			wantReport: generate.Report{
+				Failures: []generate.FailureResult{
+					{
+						Result: generate.Result{
+							Dir: "/stack",
+						},
+						Error: errors.E(eval.ErrEval),
+					},
+				},
+			},
+		},
+		{
+			// There is no way to interpolate the expression on a string template
+			name: "not available on generate_file content",
+			layout: []string{
+				"s:stack",
+			},
+			configs: []hclconfig{
+				{
+					path: "/stack",
+					add: GenerateFile(
+						Labels("expr.txt"),
+						Str("content", `generated: ${tm_hcl_expression("data")}`),
 					),
 				},
 			},

--- a/hcl/eval/eval.go
+++ b/hcl/eval/eval.go
@@ -57,7 +57,7 @@ func NewContext(basedir string) (*Context, error) {
 	}
 
 	hclctx := &hhcl.EvalContext{
-		Functions: newTmFunctions(basedir),
+		Functions: newDefaultFunctions(basedir),
 		Variables: map[string]cty.Value{},
 	}
 	return &Context{
@@ -65,18 +65,23 @@ func NewContext(basedir string) (*Context, error) {
 	}, nil
 }
 
-// SetTmVendor sets the tm_vendor function on this evaluation context.
+// AddTmVendor adds the tm_vendor function on this evaluation context.
 // The targetdir defines what tm_vendor will use to define the relative paths
 // of vendored dependencies.
 // The vendordir defines where modules are vendored inside the project.
 // The stream defines the event stream for tm_vendor, one event is produced
 // per successful function call.
-func (c *Context) SetTmVendor(
+func (c *Context) AddTmVendor(
 	targetdir project.Path,
 	vendordir project.Path,
 	stream chan<- event.VendorRequest,
 ) {
 	c.hclctx.Functions["tm_vendor"] = tmVendor(targetdir, vendordir, stream)
+}
+
+// AddTmHCLExpression adds the tm_hcl_expression function on this evaluation context.
+func (c *Context) AddTmHCLExpression() {
+	c.hclctx.Functions["tm_hcl_expression"] = tmHCLExpression()
 }
 
 // SetNamespace will set the given values inside the given namespace on the

--- a/hcl/eval/functions.go
+++ b/hcl/eval/functions.go
@@ -30,7 +30,7 @@ import (
 	"github.com/zclconf/go-cty/cty/function"
 )
 
-func newTmFunctions(basedir string) map[string]function.Function {
+func newDefaultFunctions(basedir string) map[string]function.Function {
 	scope := &tflang.Scope{BaseDir: basedir}
 	tffuncs := scope.Functions()
 
@@ -44,7 +44,6 @@ func newTmFunctions(basedir string) map[string]function.Function {
 
 	// sane ternary
 	tmfuncs["tm_ternary"] = tmTernary()
-	tmfuncs["tm_hcl_expression"] = tmHCLExpression()
 	return tmfuncs
 }
 

--- a/hcl/eval/functions_test.go
+++ b/hcl/eval/functions_test.go
@@ -136,7 +136,7 @@ func TestTmVendor(t *testing.T) {
 			ctx, err := eval.NewContext(rootdir)
 			assert.NoError(t, err)
 
-			ctx.SetTmVendor(targetdir, vendordir, events)
+			ctx.AddTmVendor(targetdir, vendordir, events)
 
 			gotEvents := []event.VendorRequest{}
 			done := make(chan struct{})
@@ -170,7 +170,7 @@ func TestTmVendor(t *testing.T) {
 				ctx, err := eval.NewContext(rootdir)
 				assert.NoError(t, err)
 
-				ctx.SetTmVendor(targetdir, vendordir, nil)
+				ctx.AddTmVendor(targetdir, vendordir, nil)
 
 				val, err := ctx.Eval(test.NewExpr(t, tcase.expr))
 				assert.NoError(t, err)

--- a/hcl/hcl_stack_test.go
+++ b/hcl/hcl_stack_test.go
@@ -21,6 +21,7 @@ import (
 	"github.com/madlambda/spells/assert"
 	"github.com/mineiros-io/terramate/errors"
 	"github.com/mineiros-io/terramate/hcl"
+	"github.com/mineiros-io/terramate/hcl/eval"
 	. "github.com/mineiros-io/terramate/test/hclutils"
 )
 
@@ -161,6 +162,42 @@ func TestHCLParserStack(t *testing.T) {
 				errs: []error{
 					errors.E(hcl.ErrTerramateSchema),
 					errors.E(hcl.ErrTerramateSchema),
+				},
+			},
+		},
+		{
+			name: "tm_vendor is not available on stack block",
+			input: []cfgfile{
+				{
+					filename: "stack.tm",
+					body: `
+						stack{
+						  name = tm_vendor("github.com/mineiros-io/terramate?ref=v2")
+						}
+					`,
+				},
+			},
+			want: want{
+				errs: []error{
+					errors.E(eval.ErrEval),
+				},
+			},
+		},
+		{
+			name: "tm_hcl_expression is not available on stack block",
+			input: []cfgfile{
+				{
+					filename: "stack.tm",
+					body: `
+						stack{
+						  name = tm_hcl_expression("ref")
+						}
+					`,
+				},
+			},
+			want: want{
+				errs: []error{
+					errors.E(eval.ErrEval),
 				},
 			},
 		},

--- a/stack/globals_test.go
+++ b/stack/globals_test.go
@@ -1445,6 +1445,19 @@ func TestLoadGlobals(t *testing.T) {
 			},
 		},
 		{
+			name:   "tm_hcl_expression is not available on globals",
+			layout: []string{"s:stack"},
+			configs: []hclconfig{
+				{
+					path: "/stack",
+					add: Globals(
+						Expr("data", `tm_hcl_expression("test")`),
+					),
+				},
+			},
+			wantErr: errors.E(globals.ErrEval),
+		},
+		{
 			name:   "global interpolating multiple lists fails",
 			layout: []string{"s:stack"},
 			configs: []hclconfig{

--- a/stack/globals_test.go
+++ b/stack/globals_test.go
@@ -54,6 +54,8 @@ func TestLoadGlobals(t *testing.T) {
 		}
 	)
 
+	t.Parallel()
+
 	tcases := []testcase{
 		{
 			name:   "no stacks no globals",
@@ -1458,6 +1460,19 @@ func TestLoadGlobals(t *testing.T) {
 			wantErr: errors.E(globals.ErrEval),
 		},
 		{
+			name:   "tm_vendor is not available on globals",
+			layout: []string{"s:stack"},
+			configs: []hclconfig{
+				{
+					path: "/stack",
+					add: Globals(
+						Expr("data", `tm_vendor("github.com/mineiros-io/terramate?ref=test")`),
+					),
+				},
+			},
+			wantErr: errors.E(globals.ErrEval),
+		},
+		{
 			name:   "global interpolating multiple lists fails",
 			layout: []string{"s:stack"},
 			configs: []hclconfig{
@@ -2435,8 +2450,11 @@ func TestLoadGlobalsErrors(t *testing.T) {
 		},
 	}
 
-	for _, tcase := range tcases {
+	for _, tc := range tcases {
+		tcase := tc
 		t.Run(tcase.name, func(t *testing.T) {
+			t.Parallel()
+
 			s := sandbox.New(t)
 			s.BuildTree(tcase.layout)
 

